tests/test_transforms.py

```diff
@@ -258,3 +262,22 @@ def transform_class(cls):
                 import UserDict
         """
         )
+
+    def test_transform_aborted_if_recursion_limited(self):
+        def transform_call(node: Call) -> Const:
+            return node
+
+        self.transformer.register_transform(
+            nodes.Call, transform_call, _looks_like_dataclass_field_call
+        )
+
+        original_limit = sys.getrecursionlimit()
+        sys.setrecursionlimit(500 if IS_PYPY else 1000)
+
+        try:
+            with self.assertWarns(
+                UserWarning, msg="try: --init-hook='import sys; sys.setrecursionlimit"
+            ):
```

Comment:
I would use something like `with pytest.warns(UserWarning, match="try: --init-hook='import sys; sys.setrecursionlimit")` here.

Reply:
Thanks, I often forget that this project uses pytest ;)